packages/blocks/stopwatch/src/app.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState, useCallback } from "react";
+import { BlockComponent } from "blockprotocol/react";
+
+type AppProps = {
+  start: Date;
+  laps: number[];
+};
+
+function formatDuration(duration: number) {
+  const MILLISECONDS = 1;
+  const SECONDS = 1000 * MILLISECONDS;
+  const MINUTES = 60 * SECONDS;
+  const HOURS = 60 * MINUTES;
+  const DAYS = 24 * HOURS;
+
+  const millis = Math.floor((duration % SECONDS) / 10)
+    .toString()
+    .padStart(2, "0");
+  const seconds = Math.floor((duration % MINUTES) / SECONDS)
+    .toString()
+    .padStart(2, "0");
+  const minutes = Math.floor((duration % HOURS) / MINUTES)
+    .toString()
+    .padStart(2, "0");
+  const hours = Math.floor((duration % DAYS) / HOURS);
+  const days = Math.floor(duration / DAYS);
+
+  if (duration >= DAYS) {
+    return `${days}:${hours}:${minutes}:${seconds}.${millis}`;
+  } else if (duration >= HOURS) {
+    return `${hours}:${minutes}:${seconds}.${millis}`;
+  } else {
+    return `${minutes}:${seconds}.${millis}`;
+  }
+}
+
+function lastLap(laps: number[]): number {
+  const last = laps.slice(-1)[0];
+  return last === undefined ? 0 : last;
+}
+
+export const App: BlockComponent<AppProps> = ({
+  entityId,
+  accountId,
+  updateEntities,
+  start = null,
+  laps = [0],
+}) => {
+  const [start_, setStart] = useState(start !== null ? new Date(start) : null);
+  const [laps_, setLaps] = useState(laps.length < 1 ? [0] : laps);
```

Comment:
Names like `start_` and `laps_` don’t communicate the intent as good as plain `camelCase` names. I guess we deal with `currentStart` or maybe we want to rename the incoming prop on the fly (`start: initialStart`).

Reply:
changed the variable names to `localStart` and `localLaps` in ce3f6df19c9c0569fc82bc852390e9688cb04c97. This reflects, that this is the same variable as in `AppProps` but local.